src/components/Produtos.js

```diff
@@ -41,6 +49,7 @@ Produtos.propTypes = {
     price: PropTypes.number.isRequired,
   })).isRequired,
   gestorDoCarrinho: PropTypes.func.isRequired,
+  guardaProdutoClicado: PropTypes.func.isRequired,
 };
```

Comment:
Olá, tudo bem? É uma sugestão pequena mas na próxima a gente pode colocar apenas um `isRequired` no final do objeto do PropTypes, ao invés de no final de cada prop.
```suggestion
Produtos.propTypes = {
  produtos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    thumbnail: PropTypes.string,
    price: PropTypes.number,
  })),
  gestorDoCarrinho: PropTypes.func,
  guardaProdutoClicado: PropTypes.func,
}.isRequired;
```

Reply:
Opa, tudo bem sim Pedro.

Entendi sua sugestão, mas ela tem um significado diferente de usar `isRequired` em todos os campos:

Da forma que eu fiz, não apenas a props `produtos` é obrigatória, mas todos os campos do objeto dentro do array. Na sua sugestão, um produto sem o campo `id`, por exemplo, seria aceito já que apenas o array `produtos` é obrigatório.

Concorda?

Valeu pela sugestão

Reply:
Fiz o teste aqui e deu certo. Muito bom, @pedrotrasfereti . Valeu pela sugestão 👏